pkg/epp/datalayer/metrics_test.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package datalayer_test
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"sigs.k8s.io/gateway-api-inference-extension/pkg/epp/datalayer"
+)
+
+func TestNewMetrics(t *testing.T) {
+	m := datalayer.NewMetrics()
+	assert.NotNil(t, m)
+	assert.NotNil(t, m.ActiveModels)
+	assert.NotNil(t, m.WaitingModels)
+
+	assert.Zero(t, m.MaxActiveModels)
+	assert.Zero(t, m.KVCacheUsagePercent)
+	assert.True(t, m.UpdateTime.IsZero())
+}
+
+func TestMetricsClone(t *testing.T) {
+	now := time.Now()
+
+	m := &datalayer.Metrics{
+		ActiveModels:            map[string]int{"modelA": 1},
+		WaitingModels:           map[string]int{"modelB": 2},
+		MaxActiveModels:         5,
+		RunningQueueSize:        3,
+		WaitingQueueSize:        7,
+		KVCacheUsagePercent:     42.5,
+		KvCacheMaxTokenCapacity: 2048,
+		UpdateTime:              now,
+	}
```

Comment:
I strongly recommend using the same testing patterns from other packages, no need to test each field, there are packages that do it (whole object comparison). see example here:
https://github.com/kubernetes-sigs/gateway-api-inference-extension/blob/d9c5c27dd1687c190743897b9f596f88fb7f40f7/pkg/epp/scheduling/framework/plugins/picker/picker_test.go#L115-L117